tests/unit/forklift/test_forms.py

```diff
@@ -10,45 +10,59 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# TODO:
+#  `tests.unit.forklift.test_legacy.TestFileUpload.test_fails_invalid_post_data`
+#  already covers the full form with specific cases.
+#  Should we remove these test cases? We will remain at 100% coverage.
+
+import hashlib
 
 import pytest
 
 from webob.multidict import MultiDict
-from wtforms.validators import ValidationError
 
 from warehouse.forklift.forms import UploadForm
 
 
 class TestUploadForm:
+    _sha256 = hashlib.sha256().hexdigest()
+    _blake2b = hashlib.blake2b(digest_size=32).hexdigest()
+
     @pytest.mark.parametrize(
         "data",
         [
             # Test for singular supported digests
             {"filetype": "sdist", "md5_digest": "bad"},
             {"filetype": "bdist_wheel", "pyversion": "3.4", "md5_digest": "bad"},
-            {"filetype": "sdist", "sha256_digest": "bad"},
-            {"filetype": "bdist_wheel", "pyversion": "3.4", "sha256_digest": "bad"},
-            {"filetype": "sdist", "blake2_256_digest": "bad"},
-            {"filetype": "bdist_wheel", "pyversion": "3.4", "blake2_256_digest": "bad"},
+            {"filetype": "sdist", "sha256_digest": _sha256},
+            {"filetype": "bdist_wheel", "pyversion": "3.4", "sha256_digest": _sha256},
+            {"filetype": "sdist", "blake2_256_digest": _blake2b},
+            {
+                "filetype": "bdist_wheel",
+                "pyversion": "3.4",
+                "blake2_256_digest": _blake2b,
+            },
             # Tests for multiple digests passing through
             {
                 "filetype": "sdist",
                 "md5_digest": "bad",
-                "sha256_digest": "bad",
-                "blake2_256_digest": "bad",
+                "sha256_digest": _sha256,
+                "blake2_256_digest": _blake2b,
             },
             {
                 "filetype": "bdist_wheel",
                 "pyversion": "3.4",
                 "md5_digest": "bad",
-                "sha256_digest": "bad",
-                "blake2_256_digest": "bad",
+                "sha256_digest": _sha256,
+                "blake2_256_digest": _blake2b,
             },
         ],
     )
     def test_full_validate_valid(self, data):
+        # `name` is required for any submission
+        data["name"] = "fake-package"
         form = UploadForm(MultiDict(data))
-        form.full_validate()
+        assert form.validate(), form.errors
```

Comment:
I don't understand the assert here. does this mean `form.validate()` and `form.errors` both are truthy? I'd assume on a valid form that form.errors would be false.

Reply:
This is leveraging `assert`s built-in behavior to allow customization of the assertion error output.

Paraphrasing:

> If form.validate() is not truthy, show an assertion error with the form.errors so we can more easily debug why it didn't validate

https://docs.python.org/3/reference/simple_stmts.html#assert